### PR TITLE
Completed the task

### PR DESCRIPTION
**Pull request by Likh-Alex**
- Created `CinemaHall` and `MovieSession` models
- Createed `CinemaHallDao` and `MovieSessionDao `
- Created `MovieSessionService` and `CinemaHallService `
- Implemented relations between `CinemaHall` and `MovieSession` models
- Added example of using all new methods in the Main class

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session
+                    .createQuery("SELECT ms FROM MovieSession ms "
+                            + "WHERE ms.id = :id "
+                            + "AND ms.showTime >= :timeFrom "
+                            + "AND ms.showTime <= :timeTo", MovieSession.class)
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java:59]: I made this with fetch joins. Don't know if it is necessary here.

File: src/main/java/com/dev/cinema/service/MovieSessionService.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.dev.cinema.model.MovieSession;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface MovieSessionService {
+    List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
+
+    MovieSession add(MovieSession session);
+}
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/cinema/service/MovieSessionService.java:11]: Why didn't you add here `getAll() `method and its implementation, if you did it in `MovieSessionDao`?

**Review by Alex-Korzun — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java:59]: As I understood from QnA - both options are valid, if model relations is established correctly everything works just fine

File: src/main/java/com/dev/cinema/service/MovieSessionService.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/cinema/service/MovieSessionService.java:11]: I thought if it is not in the requirements for the `MovieSessionService` so there is no requirement to implement it in the `MovieSessionDao` respectively. I think it is no problem to add of request)

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -1,18 +1,78 @@
...
+        yesterdayMovieSession.setCinemaHall(oldTypeCinemaHall);
+        yesterdayMovieSession.setShowTime(LocalDateTime.now().minusDays(1));
+        movieSessionService.add(yesterdayMovieSession);
+
+        List<MovieSession> availableSessions = movieSessionService
+                .findAvailableSessions(1L, LocalDate.now());
+        List<MovieSession> availableSessions1 = movieSessionService
+                .findAvailableSessions(2L, LocalDate.now().plusDays(1L));
+        List<MovieSession> availableSessions2 = movieSessionService
+                .findAvailableSessions(3L, LocalDate.now().minusDays(1L));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/Main.java]: let's use the id from the real movie objects

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

File: src/main/java/com/dev/cinema/Main.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/cinema/Main.java]: Surely, done.

**Pull request merged**